src/main/java/org/commonjava/service/metadata/handler/PromoteEventConsumer.java

```diff
@@ -38,17 +40,23 @@ public CompletionStage<Void> receive( Message<PathsPromoteCompleteEvent> message
 
         logger.info( "Got an event: {}", event );
 
-        final String keyStr = event.getTargetStore();
+        final String keyStr = event.isPurgeSource() ? event.getSourceStore() : event.getTargetStore();
```

Comment:
On normal promotion, user can also set purgeSource. So purgeSource does not always mean rollback. To be safe, in this case, just add source and its groups to the filesystem set.